src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import java.util.Map;
+
+import static org.elasticsearch.script.ScriptService.ScriptType;
+
+public class Script {
+
+    public final ScriptType type;
+    public final String lang;
+    public final String key;
```

Comment:
cacheKey would be a better name, but I've changed it back to script for now.  It just seemed a bit odd to me to have this named script when in reality the only script being passed in is for dynamic ones.  However, it could be argued that script is the key in this case, so I'm going to leave it at that.